metadash/src/components/Dashboard.vue

```diff
@@ -1,100 +1,173 @@
+              </v-card-text>
+            </draggable>
+          </v-card>
+        </v-flex>
+        <v-flex d-flex xs12 sm6 md8>
+          <v-card flat :color="cardColor">
+            <draggable :list="bigDashboards" class="dragArea" row wrap @start="drag=true" :options="{group:'board'}" @end="drag=false" :move="onMoveCallback">
+              <v-flex v-for="dashboard in bigDashboards" :key="dashboard.name">
+                <v-card>
+                  <component v-bind:is="dashboard.module"></component>
+                </v-card>
+              </v-flex>
+              <v-card-text v-show="drag">
+                <v-container style="height: 480px;" fluid fill-height>
+                  <v-layout justify-center align-center>
+                    <v-flex text-xs-center>
+                      <v-icon x-large>add</v-icon>
+                    </v-flex>
+                  </v-layout>
+                </v-container>
+              </v-card-text>
+            </draggable>
+          </v-card>
+        </v-flex>
+        <v-flex d-flex xs12 sm6 md4>
+          <v-card flat :color="cardColor">
+            <draggable :list="midDashboards" class="dragArea" row wrap @start="drag=true" :options="{group:'board'}" @end="drag=false" :move="onMoveCallback">
+              <v-flex v-for="dashboard in midDashboards" :key="dashboard.name">
+                <v-card>
+                  <component v-bind:is="dashboard.module"></component>
+                </v-card>
+              </v-flex>
+              <v-card-text v-show="drag">
+                <v-container style="height: 480px;" fluid fill-height>
+                  <v-layout justify-center align-center>
+                    <v-flex text-xs-center>
+                      <v-icon x-large>add</v-icon>
+                    </v-flex>
+                  </v-layout>
+                </v-container>
+              </v-card-text>
+            </draggable>
+          </v-card>
+        </v-flex>
+      </v-layout>
+    </v-container>
+  </v-card>
 </template>
 
 <script>
 import draggable from 'vuedraggable'
 import { Plugins } from '@/plugin'
 export default {
-  name: 'hello',
+  name: 'dashboard',
   components: { draggable },
   data () {
     return {
-      actionsFab: true,
+      menu: false,
+      fab: false,
       drag: false,
-      dashboards: []
+      backImgUrl: '',
+      uponDashboards: {
+        'upBoard0': [],
+        'upBoard1': [],
+        'upBoard2': [],
+        'upBoard3': []
+      },
+      savedDashboards: [],
+      bigDashboards: [],
+      midDashboards: []
     }
   },
   methods: {
     closePlugin (dashboard) {
       const index = this.dashboards.indexOf(dashboard)
       this.dashboards.splice(index, 1)
+    },
+    onMoveCallback (data, event) {
+      if (data.to.className === 'uponDragArea' && data.relatedContext.list.length > 0) {
+        return false
+      }
+    },
+    loadBackgroundImg () {
+      this.$mdAPI.get('/configs/BACKGROUND_IMG_URL')
```

Comment:
The config should be available in store by the time dashboard is loaded, no need to fetch again.

Reply:
okay, i will add a new helper in store.js to support get config with key